File: DailyChallenges/CPP/LeetCode_997.cpp
class Solution {
public:
    int findJudge(int n, vector<vector<int>>& trust) {
        vector<bool> t(n+1); 
        unordered_map<int, unordered_set<int>> m;
        for (auto& a: trust){
            t[a[0]] = true;
            if (!t[a[1]])
                m[a[1]].insert(a[0]);
        }
        int cnt = 0, judge = 0;
        for (int i = 1; i<=n; ++i){
            if (!t[i] && m[i].size() == n-1){
                ++cnt; 
                judge = i;
            }
        }
        return (cnt == 1) ? judge : -1;
    }
};

File: DailyChallenges/Scala/LeetCode_997.scala
object Solution {
    def findJudge(n: Int, trust: Array[Array[Int]]): Int = {
        if (n==1 && trust.length == 0) return 1
        val allTrusts = trust.groupMap(a => a(1))(a => a(0)).filter(_._2.distinct.length == n-1).keys.toSet
        val judges = allTrusts -- trust.map(_(0)).toSet
        if (judges.size == 1) judges.head else -1 
    }
}
